pkg/acsengine/engine.go

```diff
@@ -723,20 +725,34 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 		"GetDCOSMasterCustomData": func() string {
 			masterProvisionScript := getDCOSMasterProvisionScript()
 			masterAttributeContents := getDCOSMasterCustomNodeLabels()
+			masterPreprovisionExtentsion := ""
+			if cs.Properties.MasterProfile.PreprovisionExtension != nil {
+				masterPreprovisionExtentsion += "\n"
+				masterPreprovisionExtentsion += makeMasterExtensionScriptCommands(cs)
+			}
+
 			str := getSingleLineDCOSCustomData(
 				cs.Properties.OrchestratorProfile.OrchestratorType,
 				cs.Properties.OrchestratorProfile.OrchestratorRelease,
-				cs.Properties.MasterProfile.Count, masterProvisionScript, masterAttributeContents)
+				cs.Properties.MasterProfile.Count, masterProvisionScript,
+				masterAttributeContents, masterPreprovisionExtentsion)
 
 			return fmt.Sprintf("\"customData\": \"[base64(concat('#cloud-config\\n\\n', '%s'))]\",", str)
 		},
 		"GetDCOSAgentCustomData": func(profile *api.AgentPoolProfile) string {
 			agentProvisionScript := getDCOSAgentProvisionScript(profile)
 			attributeContents := getDCOSAgentCustomNodeLabels(profile)
+			agentPreprovisionExtentsion := ""
```

Comment:
typo: `agentPreprovisionExtentsion` should be `agentPreprovisionExtension` (remove extra `t`)